velox/functions/prestosql/DateTimeFunctions.h

```diff
@@ -466,6 +470,32 @@ struct TimestampMinusIntervalDayTime {
   }
 };
 
+template <typename T>
+struct TimestampAtTimezoneFunction : public TimestampWithTimezoneSupport<T> {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  FOLLY_ALWAYS_INLINE void call(
+      out_type<TimestampWithTimezone>& result,
+      const arg_type<Timestamp>& ts,
+      const arg_type<Varchar>& timeZone) {
+
+    auto timeZoneStr = std::string_view(timeZone.data(), timeZone.size());
+    auto zone = date::locate_zone(timeZoneStr);
+    std::chrono::system_clock::time_point tp{std::chrono::seconds{ts.getSeconds()}};
+
+    // Africa/Abidjan has offset +00:00 -> use this timezone to establish UTC as starting offset
+    auto startZonedTime = date::make_zoned("Africa/Abidjan", tp);
```

Comment:
why not GMT?

Reply:
I was using the terms interchangeably as they are practically equivalent in this context - can replace with GMT. My point was that the input time is to be interpreted at GMT, so the Africa/Abidjan timezone, with offset of +00:00, is used to construct a starting timepoint based on the input timestamp.

Reply:
can we not use the actual string "GMT" for clarity?